### PR TITLE
Rspec unit tests PR

### PR DESCRIPTION
**Pull request by mertkantaroglu**
Hello,

In this branch me and @josuerushanika;

- Added below unit tests:
  - Book
  - Person
  - Student
  - Teacher
  - Classroom
  - Rental
  - Decorator 
  - Capitalize decorator
  - Trimmer decorator
- Fixed linter errors

Looking forward to hearing from you.

Thanks.

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require './book'
```
Comment (gemmen29) [inline spec/book_spec.rb:1]: - Good Job on implementing tests for all files asked but there are still some methods that didn't have any test Kindly add tests for the following methods as per this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md#:~:text=Add%20unit%20tests%20for%20all%20the%20methods%20in%20all%20the%20classes%3A). They asked to add tests for all methods in all classes. The missing methods are:
   - [x] `add_rental` and `to_h` in the Book class.
   - [x] `add_rental` and `correct_name` in the Person class.
   - [x] `add_classroom` and `to_h` in the Student class.
   - [x] `to_h` in the Teacher class.
   - [x] `add_student` in the Classroom class.
   - [x] `to_h` in the Rental class.
   - [x] `correct_name` in the Nameable class.
   
for some of the previous methods, you test it manually without using the function. Kindly use the function instead.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/book_spec.rb (hunk 1 above)
Comment (mertkantaroglu) [reply spec/book_spec.rb:1]: Thanks for the info @gemmen29 . We added required tests to the project.

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @mertkantaroglu and @josuerushanika    :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mertkantaroglu** [comment]
Thank you @hiromataba 👍

**Pull request merged**